mvc/models/model-binding.rst

```diff
@@ -1,8 +1,89 @@
+
+When a parameter is bound, model binding stops looking for parameters with that name and it moves on to bind the next parameter. If binding fails, MVC does not throw an error. The error is recorded in the ``ModelState.Errors`` collection, and model binding attempts to bind to the next parameter or property. You can query for model state errors by checking the ``ModelState.IsValid`` method. 
+
+.. Note:: The ``ModelState.Errors`` collection is a property of the controller's ``ModelStateDictionary`` object, containing name-value pairs. It's rarely necessary to query this collection yourself; if there are items in this collection, then ``ModelState.IsValid`` returns ``false``. 
+
+Additionally, there are some special data types that MVC must consider when performing model binding:
+
+- ``IFormFile``: Uploaded files that are part of the HTTP request.
+- ``IFormCollection``: The collection of key-value pairs that represent the form data.
+- ``CancelationToken``: Used to cancel activity in asynchronous controllers.
+
+Once model binding is complete, `validation <https://docs.asp.net/projects/mvc/en/latest/models/validation.html>`_ occurs. Default model binding works great for the vast majority of development scenarios. It is also extensible so if you have unique needs you can customize the built-in behavior.  
+
+Customize model binding behavior with attributes 
+--------------------------------------------------------
+MVC contains several attributes that you can use to direct its default model binding behavior to a different source. For example, you can specify whether binding is required for a field, or if it should never happen at all by using the ``[BindRequired]`` or ``[BindNever]`` attributes. Alternatively, you can override the default value, and specify the model binder's data source. Below is a list of model binding attributes:
+
+- ``[BindRequired]``: This attribute forces a model state error if binding cannot occur.
+- ``[BindNever]``: Tells the model binder to never bind to this parameter.
+- ``[FromHeader]``, ``[FromQuery]``, ``[FromRoute]``, ``[FromForm]``: Use these to specify the exact binding source you want to apply.
+- ``[FromServices]``: This attribute uses `dependency injection <https://docs.asp.net/en/latest/fundamentals/dependency-injection.html>`_ to bind parameters from services.
+- ``[FromBody]``: Use the configured formatters to bind data from the request body. The formatter is selected based on content type of the request.
+- ``[ModelBinder]``: Used to override the default model binder, binding source and name.
+
+Parameter attributes are very helpful tools when you need to override the default behavior of model binding.
```

Comment:
> Parameter attributes

Just talk about "attributes" throughout this document since almost all can be applied to properties, some can be applied to types, and only a few can be applied to parameters. (`[FromServices]` is a rare case -- it can only be applied to parameters. But that's not worth highlighting.)